### PR TITLE
rfb: add RFB parser - v10

### PR DESCRIPTION
**Pull request by satta**
Previous PR: #4695 

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [#3501](https://redmine.openinfosecfoundation.org/issues/3501)

Describe changes (from previous PR):
- Request correct amount of content from app-layer buffer, addressing @victorjulien's concern from the previous PR.

### CODE REVIEW

File: rust/src/rfb/rfb.rs [hunk 1]
```diff
@@ -0,0 +1,810 @@
...
+                            current = rem;
+
+                            let chosen_security_type = request.security_type;
+                            match chosen_security_type {
+                                2 => self.state = parser::RFBGlobalState::TCVncChallenge,
+                                1 => self.state = parser::RFBGlobalState::TSClientInit,
+                                _ => return AppLayerResult::err(),
+                            }
+
+                            match self.get_current_tx() {
```
[the remark this answers is not in the document]
Comment (victorjulien) [reply rust/src/rfb/rfb.rs]: Will run this PR through my QA now.

[the remark this answers is not in the document]
Comment (victorjulien) [reply rust/src/rfb/rfb.rs]: QA is happy.

[the remark this answers is not in the document]
Comment (satta) [reply rust/src/rfb/rfb.rs]: Nice to hear that! Will address the formatting issue soon and open a new PR.

**Comment by satta** [comment]
New PR: #4708

**Comment by victorjulien** [comment]
Merged through #4710, thanks @satta and @fhonza !

**Comment by victorjulien** [comment]
I merge this one and not #4708 because I had already added it to my merge branch, with some changes (to fixup keywords after #4693 )

**Pull request closed without merging**